apps/condo/domains/meter/utils/date.utils.js

```diff
@@ -0,0 +1,76 @@
+const dayjs = require('dayjs')
+const customParseFormat = require('dayjs/plugin/customParseFormat')
+const { isEmpty, isString } = require('lodash')
+
+const { SYMBOLS_TO_CUT_FROM_DATES_REGEXP, DATE_PARSING_FORMATS } = require('@condo/domains/meter/constants/registerMetersReadingsService')
+
+dayjs.extend(customParseFormat)
+
+/**
+* @param {string} dateStr
+* @return {undefined|string}
+*/
+function sanitizeDateStr (dateStr) {
+    if (isEmpty(dateStr) || !isString(dateStr)) {
+        return undefined
+    }
+
+    const sanitizedDateStr = dateStr.replace(SYMBOLS_TO_CUT_FROM_DATES_REGEXP, '')
+    if (isEmpty(sanitizedDateStr)) {
+        return undefined
+    }
+
+    return sanitizedDateStr
+}
+
+function monthIsLoweOrEqualTwelve (dateStr) {
+    const [dateData] = dateStr.split(/T|\s/, 1)
```

Comment:
??

Reply:
just use `dayjs` for that, it should support array of formats

Reply:
don't you hard-coded (hand-written) regex for dates, it'll be hard to support it later

Reply:
Why not datejs

Some users send dates like 2024.28.05, expecting us to recognize, that 28 in the middle is day, and 05 at the end is month. That looks okey to detect. But they also want us to understand, that 2024.05.01 is not 5th february, it's 1st may, so it seems impossible to understand correctly. I checked date formats in tables, everywhere in date formats month is placed in the middle of date. Our supported formats have month only in the middle, if day present. So i decided that it's better to throw error, than trying to understand what user wants.

Yes, dayjs with array of formats is already being used, but in case of 2024.28.05 it thinks, that 28 is a month and converts overflow to year (2024.28.05 -> 2026.01.05 or something like that).

So i think that explicit check for month is required, and errors on wrong month position are required

Reply:
We should accept only ISO standard dates. `2024.28.05` should return error it's not ISO format